tools/ports/__init__.py

```diff
@@ -55,10 +58,17 @@ def read_ports():
         utils.exit_with_error('duplicate port variant: %s' % variant)
       port_variants[variant] = (port.name, extra_settings)
 
+    for key, default in port.settings.items():
+      if key in port_settings:
+        utils.exit_with_error('duplicate setting: %s' % key)
```

Comment:
Maybe just `assert key not in port_settings 'duplicate setting: %s' % key` since this should really never happen.

Hopefully if we can switch to embuilder then non-legacy ports should not have any settings to add.